### DISCUSSION TITLE
ST4000+

### DISCUSSION THREAD

**Discussion by Gmeloche** [AK-Homberger/Seatalk-Autopilot-Remote-Control#6] [General]
I use your ArduinoPilotMicro433WindNG and modified your code and add on a YK04 FOB to it and it received properly the hit keys input up to sending the datagram but the ST4000+ doesn't react. For the moment I want to known has I have the unit sit on my test table with no data input but the serial interface should it work or I have to have some kind of data connected to it. Of course I sending the Auto commend to it but no action from Standby to Auto. If I press Auto on the unit it will beep and saying No Data. For now I use Mega 2560 TX1 RX1 IDE 1.8.13 on Window10. If you are interest I could send you the .ino files, let me known.
Thanks
Gilles

**Comment by AK-Homberger** [comment]
Hi,

I tested the code originally on an ST4000+ and it worked perfectly.

But the code is not designed for an Mega 2560. And the Seatalk protocol requires the specific 9-Bit logic. Dis you install the 9-Bit patch sucessfully?

And did you use the latest code from this GitHub repository?

Regards,
Andreas

**Comment by Gmeloche** [comment]
Thanks for your reply, the 9 bits just works fine, see picture, I use your latest code, I use a YK04 FOB and it just works fine and I could create lot of key entries, see the original file here which I integrate and modify with your code, on line 94 I change it to true to see if it will transfer the whole sequence as otherwise I have a constant collisions, I use button A to send datagram Auto for now. I use 2560 for convenience right now as it was use in front of me for something else, I try to understand why I could not use it, but I have lots of other MCU like the one you use as well as ESP32, ESP8266, etc.

But..but my main question right now is the ST4000+ should work if it is sit on my lab table and only plug on your serial interface??. If I do send the datagram AUTO???

I just discover lately your nice works in the sailing environment...congratulation.

Have a good wind.

 

Gilles Meloche

438 348-5254

**Comment by AK-Homberger** [comment]
The Sketch is testing this Seatalk function by itself because it has to receive what it sends to detect collisions. 

So it does not look like your 9 bit part is working. Either hardware or software.

**Comment by Gmeloche** [comment]
Thanks, I see what you means, I am in a process right now for a sail trip for the next 5-6 weeks on Great Lake, so I put this project on idle for now. Thanks again and have a good one.

 

Gilles Meloche

438 348-5254
